### ISSUE TITLE
Frequent processTable() usage causes "Too many open files" exception

### ISSUE THREAD

**Issue by SyRenity** [jezhumble/javasysmon#7]
As the passwdFile hanlder from (passwdFile = new FileInputStream("/etc/passwd") is not closed anywhere, and rather waiting for the GC, each call to this functions opens hundreds of held descriptors to "/etc/passwd".

Running this command more often the once per 5 min, quickly depletes the available amount of descriptors and causes "Too many open files" error.

The typical descriptors list looks like this:
java    22309 evgeni 1017r   REG                9,2     2565  4426422 /etc/passwd
java    22309 evgeni 1018r   REG                9,2     2565  4426422 /etc/passwd
java    22309 evgeni 1019r   REG                9,2     2565  4426422 /etc/passwd
java    22309 evgeni 1020r   REG                9,2     2565  4426422 /etc/passwd
java    22309 evgeni 1021r   REG                9,2     2565  4426422 /etc/passwd
java    22309 evgeni 1022r   REG                9,2     2565  4426422 /etc/passwd
java    22309 evgeni 1023r   REG                9,2     2565  4426422 /etc/passwd

This issue is easily solved by closing the handler as following (quick fix, better solution most welcome):

```
public HashMap parse() {
    FileInputStream passwdFile = null;

    try {
        passwdFile = new FileInputStream("/etc/passwd");
        BufferedReader reader = new BufferedReader(new InputStreamReader(passwdFile, "UTF-8"));

        HashMap map = parse(reader);
        passwdFile.close();

        return map;
    } catch (IOException e) {
        System.err.println("Error reading password file: " + e.getMessage());

        try
        {
        if(passwdFile != null)
            passwdFile.close();
        }
        catch (Exception e2) {
            // TODO: handle exception
        }

        return new HashMap();
    }
}
```

**Comment by umairamin** [comment]
I was also getting same issue and closing readers solved the problem.

**Comment by SyRenity** [comment]
Yep, that the ticket.

On Fri, Jan 6, 2012 at 7:01 PM, umairamin <
reply@reply.github.com

**Comment by alankstewart** [comment]
I submitted a pull request with a fix for this

**Comment by jezhumble** [comment]
Fixed in 78f02f04442bc42ff99cfe2c06fa2d1a520ab1b6 thanks to @alankstewart
